draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -136,6 +136,89 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
 
 ## Option 2: Authentication Based on HTTP Message Signatures
 
+This option uses the WIMSE Identity Token (ref TBD) to sign the request and optionally, the response.
+This specification defines a profile of the Message Signatures specification {{!RFC9421}}.
+
+The request is signed as per {{RFC9421}}. The following derived components MUST be signed:
+
+* `@method`
+* `@request-target`
+
+In addition, the following headers MUST be signed when they exist:
+
+* `Content-Type`
+* `Content-Digest`
+* `Authorization`
+* `Txn-Token` {{?I-D.ietf-oauth-transaction-tokens}}
+* TBD that includes the WIT
+
+If the response is signed, the following components MUST be signed:
+
+* `@status`
+* `@method;req`
+* `@request-target;req`
+* `Content-Type` if it exists
+* `Content-Digest` if it exists
+* TBD that includes the WIT
+
+For both requests and responses, the following signature parameters MUST be included:
+
+* `created`
+* `expires` - expiration MUST be short, e.g. on the order of minutes. The WIMSE architecture will provide seperate
+mechanisms in support of long-lived compute processes.
+* `nonce`
+* `tag` - the value for implementations of this specification is `wimse-service-to-service`
+
+Since the signing key is sent along with the message, the `keyid` parameter SHOULD NOT be used.
+
+It is RECOMMENDED to include only one signature with the HTTP message.
+If multiple ones are included, then the signature label included in both the `Signature-Input` and `Signature` headers SHOULD
+be `wimse`.
+
+A sender MUST ensure that each nonce it generates is unique, at least among messages sent to the same recipient.
+To detect message replays,
+a recipient MAY reject a message (request or response) if a nonce is repeated.
+
+To promote interoperability, the `ecdsa-p256-sha256` signing algorithm MUST be implemented
+by general purpose implementations of this spec.
+
+OPEN ISSUE: do we use the `Accept-Signature` field to signal that the response must be signed?
+
+Following is a non-normative example of a signed request and a signed response, using the keys mentioned in Section TBD.
+
+~~~ http
+GET /gimme-ice-cream?flavor=vanilla HTTP/1.1
+Host: example.com
+Authorization: Basic c3BpZmZlOi8vcmVhbG0uZXhhbXBsZS5jb20vc3ZjMTphYmMxMjM=
+Signature: wimse=:L0xn/2/XncJ0QYNYuwvPsDGkJ6Gbe+rFA9soJlDfXcsqEfC7enVXFIHqCGJM7gtG6kukZUw0j/YaSXmDOiaZCQ==:
+Signature-Input: wimse=("@method" "@request-target" "authorization");created=1718102553;expires=1718102853;nonce="abcd1111";tag="wimse-service-to-service"
+
+~~~
```

Comment:
Shouldn't the example include the WIT?

Reply:
@jsalowey As above: yes when the header is defined.